src/operator/nn/dropout-inl.h

```diff
@@ -253,7 +253,19 @@ class DropoutOp {
                            const TBlob &mask,
                            const TBlob &out) {
       Stream<xpu> *s = ctx.get_stream<xpu>();
```

Comment:
not directly related to the changes in this PR: there's no chance for this to run in CPU so the xpu template here isn't necessary.

Reply:
yes, and if we can access CPU here it will be much easier, there is a getRandInt() for CPU that directly returns an unsigned number.